aspnetcore/fundamentals/websockets.md

```diff
@@ -74,18 +102,30 @@ Object name: 'HttpResponseStream'.
 
 If you're using a background service to write data to a WebSocket, make sure you keep the middleware pipeline running. Do this by using a <xref:System.Threading.Tasks.TaskCompletionSource%601>. Pass the `TaskCompletionSource` to your background service and have it call <xref:System.Threading.Tasks.TaskCompletionSource%601.TrySetResult%2A> when you finish with the WebSocket. Then `await` the <xref:System.Threading.Tasks.TaskCompletionSource%601.Task> property during the request, as shown in the following example:
 
-:::code language="csharp" source="websockets/samples/6.x/WebSocketsSample/Snippets/Program.cs" id="snippet_AcceptWebSocketAsyncBackgroundSocketProcessor":::
+:::code language="csharp" source="websockets/samples/7.x/WebSocketsSample/Snippets/Program.cs" id="snippet_AcceptWebSocketAsyncBackgroundSocketProcessor":::
 
 The WebSocket closed exception can also happen when returning too soon from an action method. When accepting a socket in an action method, wait for the code that uses the socket to complete before returning from the action method.
 
 Never use `Task.Wait`, `Task.Result`, or similar blocking calls to wait for the socket to complete, as that can cause serious threading issues. Always use `await`.
 
+## Add HTTP/2 WebSockets support for existing controllers
+
+.NET 7 introduced Websockets over HTTP/2 support for Kestrel, the SignalR JavaScript client, and SignalR with Blazor WebAssembly. HTTP/2 WebSockets use CONNECT requests rather than GET.  The previous controller action method example can be modified to use CONNECT by defining a `HttpConnect` attribute class and applying that attribute to the controller method.
+
+The following is an example custom attribute routing class named `HttpConnectAttribute`:
+
+[!code-csharp[](~/fundamentals/websockets/samples/7.x/WebSocketsSample/Controllers/HttpConnectAttribute.cs)]
+
+The custom `[HttpConnect]` attribute can be added to a controller action to enable Websocket support over HTTP/2:
```

Comment:
I think there should be a sentence describing that it should be added **in addition** to the `HttpGet` attribute, otherwise you will only support HTTP/2 websockets and not HTTP/1 websockets.